scala/private/toolchain_deps/toolchain_dep_rules.bzl

```diff
@@ -0,0 +1,29 @@
+load("@io_bazel_rules_scala//scala:providers.bzl", _ScalacProvider = "ScalacProvider")
+load("//scala/private/toolchain_deps:toolchain_deps.bzl", "expose_toolchain_deps", "java_info_for_deps")
+
+def _scala_toolchain_deps(ctx):
+    _toolchain_type = "@io_bazel_rules_scala//scala:toolchain_type"
+    from_classpath = ctx.attr.from_classpath
+
+    scalac_provider = ctx.toolchains[_toolchain_type].scalac_provider_attr[_ScalacProvider]
+    classpath_deps = getattr(scalac_provider, from_classpath)
+    return java_info_for_deps(classpath_deps)
+
+scala_toolchain_deps = rule(
+    implementation = _scala_toolchain_deps,
+    attrs = {
+        "from_classpath": attr.string(mandatory = True),
```

Comment:
`ScalacProvider` comes from the existing scala toolchain, and it would be a breaking change for some users if it gets modified. I would like, if possible, to have no breaking changes with the introduction of this toolchains infra. If it looks something that needs to be refactored into unified version, I think it's better to have a separate PR, which would be easier to revert if that changes becomes problematic.

Reply:
Got it, thanks! Seems reasonable.

Reply:
Separate PR sounds very reasonable while also important since I think having multiple mental models would make things really hard